packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,311 @@
+  <KPop
+    class="analytics-timepicker"
+    placement="bottomStart"
+    width="auto"
+    hide-caret
+  >
+    <KButton
+      :is-rounded="false"
+      size="large"
+      class="timepicker-input"
+    >
+      {{ displayedRange }}
+    </KButton>
+    <template #content>
+      <KSegmentedControl
+        v-if="hasRelativeTimeframes"
+        v-model="mode"
+        :options="[
+          { label: 'Custom', value: 'custom' },
+          { label: 'Relative', value: 'relative' }
+        ]"
+        class="w-100 mb-5"
+        data-testid="analytics-time-toggle"
+      />
+      <DatePicker
+        v-if="showCalendar"
+        :attributes="attrs"
+        :model-config="modelConfig"
+        v-model="selectedCalendarRange"
+        mode="dateTime"
+        is-range
+        is-expanded
+      />
+      <div
+        v-else-if="!showCalendar && hasRelativeTimeframes"
+        class="d-flex flex-column"
+      >
+        <div
+          v-for="(item, index) in timePeriods"
+          :key="`section-${index}`"
+          class="timeframe-section d-flex flex-column"
+        >
+          <h4>{{ item.section }}</h4>
+          <div class="timeframe-buttons d-flex flex-row justify-content-start">
+            <KButton
+              v-for="(timeFrame, index) in item.values"
+              :key="`time-${timeFrame.timeframeLength}-${index}`"
+              :is-rounded="false"
+              :class="{'selected-option': timeFrame.timeframeText === selectedTimeframe.timeframeText}"
+              :data-testid="'select-timeframe-' + timeFrame.timeframeLength"
+              appearance="outline"
+              size="medium"
+              @click="changeTimeframe(timeFrame)"
+            >
+              {{ timeFrame.timeframeText }}
+            </KButton>
+          </div>
+        </div>
+      </div>
+    </template>
+    <template #footer>
+      <div class="d-flex justify-content-end">
+        <KButton
+          :is-rounded="false"
+          size="medium"
+          appearance="btn-link"
+          @click="clearSelection()"
+        >
+          Clear
+        </KButton>
+        <KButton
+          :disabled="submitDisabled"
+          :is-rounded="false"
+          size="medium"
+          appearance="btn-link"
+          @click="submitTimeFrame()"
+        >
+          Apply
+        </KButton>
+      </div>
+    </template>
+  </KPop>
+</template>
+
+<script>
+// import { DatePicker } from 'v-calendar'
+import KButton from '@kongponents/kbutton/KButton.vue'
+import KPop from '@kongponents/kpop/KPop.vue'
+import KSegmentedControl from '@kongponents/ksegmentedcontrol/KSegmentedControl.vue'
+import { format, fromUnixTime, getUnixTime, roundToNearestMinutes } from 'date-fns'
+
+export default {
+  name: 'KTimePicker',
+  components: {
+    KButton,
+    KPop,
+    KSegmentedControl,
+    DatePicker: () => import('v-calendar/lib/components/date-picker.umd')
+  },
+  props: {
+    defaultMessage: {
+      type: String,
+      required: true
```

Comment:
Instead of requiring users to pass a `defaultMessage`, we could have a default value for this prop so users don't have to pass it every time:

```js
defaultMessage: {
  type: String,
  required: false,
  default: 'Please select a date or time range'
},
```